Detectors/MUON/MCH/Raw/Decoder/src/DataDecoder.cxx

```diff
@@ -156,21 +193,31 @@ void DataDecoder::decodeBuffer(gsl::span<const std::byte> page)
       return;
     }
 
-    Digit::Time time;
-    time.sampaTime = sc.sampaTime;
-    time.bunchCrossing = sc.bunchCrossing;
-    time.orbit = orbit;
+    SampaInfo sampaInfo;
+    sampaInfo.chip = channel / 32;
+    sampaInfo.ds = dsElecId.elinkId();
+    sampaInfo.solar = dsElecId.solarId();
+    sampaInfo.sampaTime = sc.sampaTime;
+    sampaInfo.bunchCrossing = sc.bunchCrossing;
+    sampaInfo.orbit = mOrbit;
 
-    mOutputDigits.emplace_back(o2::mch::Digit(deId, padId, digitadc, time, sc.nofSamples()));
+    mOutputDigits.emplace_back(o2::mch::Digit(deId, padId, digitadc, 0, sc.nofSamples()));
+    mSampaInfos.emplace_back(sampaInfo);
 
     if (mDebug) {
-      std::cout << "DIGIT STORED:\nADC " << mOutputDigits.back().getADC() << " DE# " << mOutputDigits.back().getDetID() << " PadId " << mOutputDigits.back().getPadID() << " time " << mOutputDigits.back().getTime().sampaTime << std::endl;
+      std::cout << "DIGIT STORED:\nADC " << mOutputDigits.back().getADC() << " DE# " << mOutputDigits.back().getDetID() << " PadId " << mOutputDigits.back().getPadID() << " time " << mSampaInfos.back().sampaTime << std::endl;
     }
     ++ndigits;
   };
 
   const auto dumpDigits = [&](bool bending) {
```

Comment:
The decodePage function is quite big and not easily readable. I would extract the dumpXXX lambas into separate (member) functions to alleviate this.

Reply:
Ok.